contracts/governor/src/storage.rs

```diff
@@ -1,26 +1,344 @@
+///
+/// ### Arguments
+/// * `proposal_id` - The id of the proposal to fetch
+pub fn get_proposal(e: &Env, proposal_id: &u32) -> Option<Proposal> {
+    let key = GovernorDataKey::Proposal(*proposal_id);
+    get_temporary_default(&e, &key, None, LEDGER_THRESHOLD_SHARED, LEDGER_BUMP_SHARED)
+}
+
+/// Store proposal in vec of proposals
+///
+/// ### Arguments
+/// * `proposal` - The proposal to store
+pub fn set_proposal(e: &Env, proposal_id: &u32, proposal: &Proposal) {
+    let key = GovernorDataKey::Proposal(*proposal_id);
+    e.storage()
+        .temporary()
+        .set::<GovernorDataKey, Proposal>(&key, proposal);
+    e.storage()
+        .temporary()
+        .extend_ttl(&key, LEDGER_THRESHOLD_SHARED, LEDGER_BUMP_SHARED);
+}
+
+// Get the proposal status for proposal at `proposal_id`
+///
+/// ### Arguments
+/// * `proposal_id` - The proposal status id
+pub fn get_proposal_status(e: &Env, proposal_id: &u32) -> ProposalStatus {
+    let key = GovernorDataKey::ProposalStatus(*proposal_id);
+    get_temporary_default::<GovernorDataKey, ProposalStatus>(
+        &e,
+        &key,
+        ProposalStatus::Pending,
+        LEDGER_THRESHOLD_SHARED,
+        LEDGER_BUMP_SHARED,
+    )
+}
+
+/// Set the proposal status for proposal at `proposal_id`
+///
+/// ### Arguments
+/// * `proposal_id` - The proposal id
+pub fn set_proposal_status(e: &Env, id: &u32, proposal_status: &ProposalStatus) {
+    let key = GovernorDataKey::ProposalStatus(*id);
+    e.storage()
+        .temporary()
+        .set::<GovernorDataKey, ProposalStatus>(&key, &proposal_status);
+    e.storage()
+        .temporary()
+        .extend_ttl(&key, LEDGER_THRESHOLD_SHARED, LEDGER_BUMP_SHARED);
+}
+
+/********** Vote **********/
+/// Set the voter status of `voter` for proposal at `proposal_id` to `status`
+///
+/// ### Arguments
+/// * `voter` - The address of the voter
+/// * `proposal_id` - The proposal id
+/// * `status` - The status the user chose
+///                 - 0 to vote abstain
+///                 - 1 to vote against
+///                 - 2 to vote for
+pub fn set_voter_status(e: &Env, voter: &Address, proposal_id: &u32, status: &u32) {
+    let key = GovernorDataKey::VoterStatus(VoterStatusKey {
+        voter: voter.clone(),
+        proposal_id: *proposal_id,
+    });
+    e.storage()
+        .temporary()
+        .set::<GovernorDataKey, u32>(&key, &status);
+    e.storage()
+        .temporary()
+        .extend_ttl(&key, LEDGER_THRESHOLD_SHARED, LEDGER_BUMP_SHARED);
+}
+
+/// Get the voter status of `voter` for proposal at `proposal_id`
+///
+/// ### Arguments
+/// * `voter` - The address of the voter
+/// * `proposal_id` - The proposal id
+pub fn get_voter_status(e: &Env, voter: &Address, proposal_id: &u32) -> Option<u32> {
+    let key = GovernorDataKey::VoterStatus(VoterStatusKey {
+        voter: voter.clone(),
+        proposal_id: *proposal_id,
+    });
+    get_temporary_default(&e, &key, None, LEDGER_THRESHOLD_SHARED, LEDGER_BUMP_SHARED)
```

Comment:
can we use a new threshold for proposals?

say `PROPOSAL_THRESHOLD` and `PROPOSAL_BUMP` - these can probably be defaulted to 30 and 31 days, respectively.

10 days is def not long enough